mailipy/send.py

```diff
@@ -17,6 +17,8 @@ def main():
     parser.add_argument("outbox", help="the folder where to access the emails that should be sent")
     parser.add_argument("sent", nargs="?", default="sent", help="the folder where to move the emails sent (default: sent)")
     parser.add_argument("--continue", help="continue sending, even if the 'sent' folder is not empty", action="store_true", dest="continue_")
+    parser.add_argument("--password", "-p", help="password for the SMTP server", action="store")
```

Comment:
Using this would mean saving the password in the bash_history, wouldn't it?

If we're OK with this, then the most natural way would be to provide the password like we do with the username (as a positional argument). What do you think?

Anyway, if we think it's worthwhile to "force" users to not log passwords in their bash_history, then maybe the best solution here would be to accept a flag with the path of the file where to read the password from (e.g. `--read-password-from`) and this flag would also replace `--password-stdin`, because we could support `-` as stdin.

Reply:
> Using this would mean saving the password in the bash_history, wouldn't it?

It depends, if you just use `-p thepassword` yes. If you use `-p $(cat password.txt)` no.

> If we're OK with this, then the most natural way would be to provide the password like we do with the username (as a positional argument). What do you think?

That would be a breaking change, at least if you don't want to ask for the password as last argument. I see no advantage over `-p`.

> Anyway, if we think it's worthwhile to "force" users to not log passwords in their bash_history

I'm not that in favor of forcing it, for example mysql / psql support `-pthepassword`.

> then maybe the best solution here would be to accept a flag with the path of the file where to read the password from (e.g. --read-password-from) and this flag would also replace --password-stdin, because we could support - as stdin.

Well, we've already discussed it. It seems (to me at least) that most of the tools accept `-p password` and/or `--password-stdin`. `--password-file` is kinda rare:

- docker → `-p` + `--password-stdin`
- keepass → `--pw-stdin`
- azure/office365 → `-p`

There are some exceptions, the first that comes to my mind is Kubernetes secrets `--from-file`.

Reply:
OK on all fronts (although I think it would be handy to have the option to take the password from a file, I know you could redirect stdin, but it's ugly...)